integration/tests/posit/connect/test_deployments.py

```diff
@@ -0,0 +1,76 @@
+import os
+import time
+from pathlib import Path
+from posit import connect
+import requests
+
+
+class TestExtensionDeployment:
+    def setup_method(self):
+        """Set up test by creating a content placeholder."""
+        self.client = connect.Client()
+        self.content = self.client.content.create()
+
+    def teardown_method(self):
+        """Clean up any content created during the test."""
+        self.content.delete()
+        assert self.client.content.count() == 0
+
+    def test_extension_deploys(self):
+        """Test that an Extension can be deployed to Posit Connect."""
+        # Get required environment variables
+        extension_name = os.getenv("EXTENSION_NAME")
+        api_key = os.getenv('CONNECT_API_KEY')
```

Comment:
It looks like the `Makefile` handles setting up the env vars for this which is a really nice way to avoid the double checking necessary if they were set in the GitHub workflow/actions individually. 👍